### PR TITLE
ref(setup-wizard): Generate org auth token for users with one org

### LINKED ISSUE #394: Create Org token for users with one only organization
Before we make large adjustments to the wizard's login/authentication UI/UX, we can easily generate an org auth token for all users who only belong to one organization. Obviously to them, the chicken/egg problem described in #357  doesn't apply.

### PR DESCRIPTION
**Pull request by Lms24**
This PR adjusts the `SetupWizard` view to generate an org auth token if users authenticate with the wizard that only belong to one org. 

Because the org tokens are still feature-flagged/disabled by default (but enabled on Sentry.io) we need to check for the flag as well. 

Also added tests to show when we generate org tokens and when we keep using user-based tokens.  

Note: To account for users with multiple orgs, we'll need to make some further adjustments in the future. This is out of scope for this PR though.

One thing I'm not sure about is hybrid cloud. Is there anything I need to consider for this?

closes https://github.com/getsentry/sentry-wizard/issues/394

### CODE REVIEW

File: src/sentry/web/frontend/setup_wizard.py [hunk 1]
```diff
@@ -90,22 +93,43 @@ def get(self, request: HttpRequest, wizard_hash) -> HttpResponse:
...
+            ]
+            if not tokens:
+                token = ApiToken.objects.create(
+                    user=request.user,
+                    scope_list=["project:releases"],
+                    refresh_token=None,
+                    expires_at=None,
+                )
+            else:
+                token = tokens[0]
```
Comment (iambriccardo) [inline src/sentry/web/frontend/setup_wizard.py:129]:
```suggestion
	    tokens = ApiToken.objects.filter(user=request.user)
            token = next((token for token in tokens if "project:releases" in token.get_scopes()), None)
            
            if token is not None:
                token = ApiToken.objects.create(
                    user=request.user,
                    scope_list=["project:releases"],
                    refresh_token=None,
                    expires_at=None,
                )
```

**Review by iambriccardo — COMMENTED** [review]
LGTM, left some minor comments.

File: src/sentry/web/frontend/setup_wizard.py [hunk 2]
```diff
@@ -90,22 +93,43 @@ def get(self, request: HttpRequest, wizard_hash) -> HttpResponse:
...
+        can_use_org_tokens = len(orgs) == 1 and features.has(
+            "organizations:org-auth-tokens", orgs[0], actor=request.user
+        )
+
+        if can_use_org_tokens:
+            org = orgs[0]
+            token_str = generate_token(org.slug, generate_region_url())
+            token_hashed = hash_token(token_str)
+            token = OrgAuthToken.objects.create(
+                name="Sentry-Wizard-Generated-Token",
```
Comment (mydea) [inline src/sentry/web/frontend/setup_wizard.py:107]: hmm, IMHO we can drop the hyphens here, this is human readable and can contain spaces, WDYT? We could also mirror the naming schema from docs:

```suggestion
                name="Generated by Wizard",
```

or even add the date (not sure from the top of my head how to do that in python, but basically "Generated by Wizard on 2023-08-12" or something like this)

For docs we use the name: `Generated by Docs for {Project Name} on 2023-08-03`

File: src/sentry/web/frontend/setup_wizard.py (hunk 1 above)
Comment (Lms24) [reply src/sentry/web/frontend/setup_wizard.py]: Much nicer! Glad to learn some Python again for a change 😅

File: src/sentry/web/frontend/setup_wizard.py (hunk 2 above)
Comment (Lms24) [reply src/sentry/web/frontend/setup_wizard.py:107]: I like it - makes things more transparent! 
I'll change it to `Generated by Sentry Wizard on {date.today()}`. I can't specify the project though as we don't know which project will be selected yet. But just the data alone makes sense.

**Review by iambriccardo — APPROVED** [review]

File: src/sentry/web/frontend/setup_wizard.py (hunk 1 above)
Comment (iambriccardo) [reply src/sentry/web/frontend/setup_wizard.py]: Oh yeah, my bad, good catch!

**Review by mydea — APPROVED** [review]

**Pull request merged**